pallets/capacity/src/lib.rs

```diff
@@ -738,3 +746,51 @@ impl<T: Config> Replenishable for Pallet<T> {
 		false
 	}
 }
+
+impl<T: Config> StakingRewardsProvider<T> for Pallet<T> {
+	type AccountId = T::AccountId;
+	type RewardEra = T::RewardEra;
+	type Hash = T::Hash;
+
+	// Calculate the size of the reward pool for the current era, based on current staked token
+	// and the other determined factors of the current economic model
+	// Currently set to 10% of total staked token.
+	fn reward_pool_size() -> Result<BalanceOf<T>, DispatchError> {
```

Comment:
I went back and forth on how to determine the reward pool total, and finally decided that rather than trying to guess what data the RewardPoolProvider needs to determine  that, any implementation needs to "know" on its own and implement fetching what it needs.  This function would return the new reward pool size based on &lt;TBD&gt; and the rest of the RewardPoolInfo data will be knowable without the trait.